### PR TITLE
Lesson 1 3

### PR DESCRIPTION
**Pull request by brrr123**
Added new component to handle AddTodoForm to handle user input.
I'm getting a couple of errors with ESLint complaining about props validation @chyashp could you confirm that the way I'm using props is correct? 
Error:(6, 16) ESLint: 'onAddTodo' is missing in props validation (react/prop-types)

Screenshot:
image

### CODE REVIEW

File: src/components/TodoListItem.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const TodoListItem = (props) =>
+    (
+        <li>{props.item.title}</li>
```
Comment (chyashp) [inline src/components/TodoListItem.jsx:3]: Based on this instruction, ``` Update the todo object reference to come from props```
can you please remove the title property. It should be ```<li>{props.item}</li>```

File: src/components/AddTodoForm.jsx [hunk 2]
```diff
@@ -1,16 +1,23 @@
-const AddTodoForm = () => {
+const AddTodoForm = (props) => {
+
+    const handleAddTodo =event =>{
+        event.preventDefault();
+         console.log(event.target.elements[0].value);
```
Comment (chyashp) [inline src/components/AddTodoForm.jsx:5]: Based on the instruction ```  Next, retrieve the value of the title element from the event target and store it in a variable named todoTitle```
can you please add```const todoTitle = event.target.elements.title.value```

Then according to this instruction```log the value of todoTitle in the console```
can you please log```console.log(todoTitle)```

File: src/components/AddTodoForm.jsx [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+         props.onAddTodo(event.target.elements[0].value);
```
Comment (chyashp) [inline src/components/AddTodoForm.jsx:6]: can you please pass todoTitle variable here , according to the instruction ``` Inside the handleAddTodo function, invoke the onAddTodo callback prop and pass todoTitle as an argument```
```props.onAddTodo(todoTitle)```

File: src/components/AddTodoForm.jsx [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
+         event.target.elements[0].value ="";
```
Comment (chyashp) [inline src/components/AddTodoForm.jsx:7]: can you please remove the index number ```[0]``` from elements.

**Review by chyashp — COMMENTED** [review]
when you run the app, it will nice to add the screenshots from the browser in the pull request description

File: src/components/TodoListItem.jsx (hunk 1 above)
Comment (brrr123) [reply src/components/TodoListItem.jsx:3]: If I change it to   <li>{props.item}</li>  I'll get an error "Uncaught Error: Objects are not valid as a React child (found: object with keys {id, title}). If you meant to render a collection of children, use an array instead" 
Also from page 56 of the Road to React book, they follow the same pattern 
image

File: src/components/TodoList.jsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
+import TodoListItem from "./TodoListItem.jsx";
+
 const todoData =[{id:1,title:"Finish the assignment"},{id:2,title:"Cook"}, {id:3,title:"Do laundry"}];
 
-const TodoList = () =>{
-    return (
-        <ul> 
-            {todoData.map(item => <li key={item.id}>{item.title}</li>)}
+const TodoList = () => (
```
[the remark this answers is not in the document]
Comment (brrr123) [reply src/components/TodoList.jsx:5]: I've send a quick message over slack about this, I'm using an implicit return as per https://www.robinwieruch.de/javascript-fundamentals-react-requirements/#arrow-functions-in-react

File: src/components/TodoListItem.jsx (hunk 1 above)
Comment (chyashp) [reply src/components/TodoListItem.jsx:3]: You're right, it makes sense, if we pass the complete item object from TodoList, we will need to use `.title` here 👍

**Review by chyashp — APPROVED** [review]

**Pull request merged**